lib/p2p/Pool.ts

```diff
@@ -712,18 +721,14 @@ class Pool extends EventEmitter {
         reject(err);
       };
 
-      this.server!.listen(this.listenPort || 0, '0.0.0.0').on('listening', () => {
-        const { address, port } = this.server!.address();
-        this.logger.info(`p2p server listening on ${address}:${port}`);
-
-        if (this.listenPort === 0) {
-          // we didn't specify a port and grabbed any available port
-          this.listenPort = port;
-        }
-
-        this.server!.removeListener('error', listenErrHandler);
-        resolve();
-      }).on('error', listenErrHandler);
+      this.interfaces!.forEach((uri) => {
+        this.server!.listen(uri.port, uri.host).on('listening', () => {
+          const { address, port } = this.server!.address();
+          this.logger.info(`p2p server listening on ${address}:${port}`);
+          resolve();
+        }).on('error', listenErrHandler);
```

Comment:
It looks like we are adding the `listenErrHandler` once for each interface. You can add it once before starting the loop with `this.server!.on('error', listenErrHandler);`.